cmd/sysinfo-fs.go

```
@@ -0,0 +1,145 @@
+/*
+ * Minio Cloud Storage, (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"github.com/minio/minio/pkg/lock"
+	"github.com/pbnjay/memory"
+	"github.com/shirou/gopsutil/cpu"
+	xdisk "github.com/shirou/gopsutil/disk"
+	"github.com/shirou/gopsutil/host"
+)
+
+// Creates a new format.json if unformatted.
+func createSysInfoFS(ctx context.Context, fsFormatPath string) error {
+	// Attempt a write lock on formatConfigFile `format.json`
+	// file stored in minioMetaBucket(.minio.sys) directory.
+	lk, err := lock.TryLockedOpenFile(fsFormatPath, os.O_RDWR|os.O_CREATE, 0600)
+	if err != nil {
+		return err
+	}
+	// Close the locked file upon return.
+	defer lk.Close()
+
+	fi, err := lk.Stat()
+	if err != nil {
+		return err
+	}
+	if fi.Size() != 0 {
+		// format.json already got created because of another minio process's createFormatFS()
+		return nil
+	}
+	sysinfo := newSysInfo(1)
+	sysinfo.NodesSysInfo[0], _ = localServerSystemInfo()
+	return jsonSave(lk.File, sysinfo)
+}
+
+// localServerSystemInfo - Returns the server info of this server.
+func localServerSystemInfo() (sid ServerSystemInfoData, e error) {
+
+	partitions, _ := xdisk.Partitions(true)
+
+	counter, _ := xdisk.IOCounters()
```

Why are we ignore errors here ? (can you add a comment?)

---

done